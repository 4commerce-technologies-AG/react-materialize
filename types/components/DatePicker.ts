import * as React from 'react';
import { AnyFn } from './utils';

interface I18nOptions {
  cancel?: string,
  clear?: string,
  done?: string,
  previousMonth?: string,
  nextMonth?: string,
  months?: string[];
  monthsShort?: string[];
  weekdays?: string[];
  weekdaysShort?: string[];
  weekdaysAbbrev?: string[];
}

export interface DatePickerOptions {
  autoClose?: boolean;
  format?: string;
  parse?: AnyFn;
  defaultDate?: Date;
  setDefaultDate?: boolean;
  disableWeekends?: boolean;
  disableDayFn?: AnyFn;
  firstDay?: number;
  minDate?: Date;
  maxDate?: Date;
  yearRange?: number | number[];
  isRTL?: boolean;
  showMonthAfterYear?: boolean;
  showDaysInNextAndPreviousMonths?: boolean;
  container?: React.ReactElement;
  showClearBtn?: boolean;
  i18n?: I18nOptions;
  events?: string[];
  onSelect?: (date: Date) => any;
  onOpen?: AnyFn;
  onClose?: AnyFn;
  onDraw?: AnyFn;
}

export interface DatePickerProps {
  onChange?: (date: Date) => any;
  id?: string;
  options?: DatePickerOptions;
};

/**
 * React Materialize: DatePicker
 */
declare const DatePicker: React.FC<DatePickerProps>

export default DatePicker;
