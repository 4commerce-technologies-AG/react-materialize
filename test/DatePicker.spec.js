import React from 'react';
import { shallow, mount } from 'enzyme';
import DatePicker from '../src/DatePicker';
import mocker from './helper/new-mocker';

describe('<DatePicker />', () => {
  let wrapper;

  test('renders', () => {
    wrapper = shallow(<DatePicker />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with a label', () => {
    wrapper = shallow(<DatePicker label="DatePicker label" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts external `id` value', () => {
    wrapper = shallow(<DatePicker id="ID" />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('initialises', () => {
    const datePickerInitMock = jest.fn();
    const datePickerInstanceDestroyMock = jest.fn();
    const datePickerMock = {
      init: (el, options) => {
        datePickerInitMock(options);
        return {
          destroy: datePickerInstanceDestroyMock
        };
      }
    };

    const restore = mocker('Datepicker', datePickerMock);

    beforeEach(() => {
      datePickerInitMock.mockClear();
      datePickerInstanceDestroyMock.mockClear();
    });

    afterAll(() => {
      restore();
    });

    test('uses default options if none are given', () => {
      wrapper = mount(<DatePicker />);

      expect(datePickerInitMock).toHaveBeenCalled();
    });
  });
});
