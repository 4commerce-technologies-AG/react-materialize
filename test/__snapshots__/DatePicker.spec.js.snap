// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> accepts external \`id\` value 1`] = `
<TextInput
  id="ID"
  inputClassName="datepicker"
/>
`;

exports[`<DatePicker /> renders 1`] = `
<TextInput
  id="DatePicker-mockId"
  inputClassName="datepicker"
/>
`;

exports[`<DatePicker /> renders with a label 1`] = `
<TextInput
  id="DatePicker-mockId"
  inputClassName="datepicker"
  label="DatePicker label"
/>
`;
